WalletWasabi.Backend/SendPushService.cs

```diff
@@ -0,0 +1,101 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Security.Cryptography;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using WalletWasabi.Backend.Data;
+using WalletWasabi.Backend.Models;
+using WalletWasabi.Logging;
+
+namespace WalletWasabi.Backend
+{
+	public class SendPushService
+	{
+		private readonly string KeyPath = "/home/Dan/Downloads/AuthKey_4L3728R8LJ.p8";
+		private WasabiBackendContext Context { get; set; }
+		private string _auth_key_id = "4L3728R8LJ";
+		private string _teamId = "9Z72DXKVXK"; // Chaincase LLC
+		private string _bundleId = "cash.chaincase.testnet"; // APNs Development iOS
+		private string _payload = @"{
+				'aps': {
+					'content-available': 1,
+					'alert': 'Finalising CoinJoin',
+					'sound': 'default'
+					}
+				}
+			}";
+
+		public SendPushService()
+		{
+		}
+
+		private string GenerateAuthenticationHeader()
```

Comment:
What is this?

Reply:
Apple requires a JWT Authentication header, it's complex unfortunately. I should:

- [ ] leave a comment summary for GenerateAuthenticationHeader

Reply:
Instead of doing all that manually, you can use dotnet libs:

```
var tokenHandler = new JwtSecurityTokenHandler();
            var securityToken = tokenHandler.CreateToken(tokenDescriptor);
            var token = tokenHandler.WriteToken(securityToken);
```

Reply:
I'm not sure it does, we need to sign the request with that private key in the p8 file